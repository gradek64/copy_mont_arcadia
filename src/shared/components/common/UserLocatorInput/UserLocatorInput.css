/** @define UserLocatorInput */

/* postcss-bem-linter: ignore */
.UserLocatorInput-loader .Loader-image {
  width: 20px;
  height: auto;
}

.UserLocatorInput {
  /** important required to override generic CSS */
  &-loader {
    height: 16px;
    top: 0 !important;
    left: 0 !important;
  }
  &-label {
    display: none;
  }
  &-inputContainer {
    position: relative;
    display: flex;
    height: 0;
    visibility: hidden;
    overflow: hidden;
    z-index: 1;
    transition: 0.3s ease;
    &.is-visible {
      visibility: visible;
      height: $user-locator-height;
      margin-top: $user-locator-margin-top;
    }
    &.is-predictionsListOpen {
      overflow: visible;
    }
  }
  &-predictionsList {
    position: absolute;
    width: 100%;
    top: $user-locator-height;
    left: 0;
    list-style: none;
    padding: 0;
    box-sizing: border-box;
    margin: 0;
  }
  &-predictionsListItem {
    border: solid $md-gray;
    border-width: 0 1px 1px 1px;
    background-color: $white;
    box-sizing: border-box;
    line-height: 2.8;
    text-overflow: ellipsis;
    overflow: hidden;
    white-space: nowrap;
    padding-left: 5px;
    padding-right: 5px;
    font-size: 1em;
    font-weight: 300;
    cursor: pointer;
  }
  &-active {
    background-color: $store-locator-prediction-active;
    box-shadow: 0 2px 0px 0 rgba(0, 0, 0, .2);
  }
  &-inActive {
    animation: fadeOut 0.3s;
  }
  &-queryInput {
    display: flex;
    flex-basis: $user-locator-input-width;
    font-weight: $user-locator-input-font-weight;
    border: none;
    box-sizing: border-box;
    font-size: 1em;
    height: 100%;
    outline: 0;
    overflow: hidden;
    padding: 0 $user-locator-input-padding;
    border: solid $user-locator-input-border-color $user-locator-input-border-width;
    border-radius: $input-border-radius 0 0 $input-border-radius;
    border-right: none;
    &:active, &:focus {
      border-color: $user-locator-input-border-focus-color;
    }
  }
  &-inputField {
    flex-grow: 1;
    height: 100%;
    outline: 0;
    font-size: $font-size-input;
    text-overflow: ellipsis;
    overflow: hidden;
    white-space: nowrap;
    -webkit-appearance: none;
    padding: 0;
    box-sizing: border-box;
    border: none;
    &::-webkit-search-cancel-button {
      -webkit-appearance: none;
    }
    &::-ms-clear {
      display: none;
    }
  }
  &-suffix {
    display: flex;
    flex-direction: row;
    justify-content: flex-end;
    align-items: center;
  }
  &-rightIcon {
    width: 16px;
    background-repeat: no-repeat;
    background-position: center center;
    background-size: $clear-icon-button;
  }
  &-loader {
    margin: 0;
  }
  &-clearButton {
    height: 16px;
    width: 16px;
    background-image: url('public/common/images/location-clear-icon.svg');
  }
  &-currentLocationButton {
    background-image: url('public/common/images/location-generic-off.svg');
    width: 16px;
    height: 16px;
    margin-left: $user-locator-input-padding;
    &.is-located {
      background-image: url('public/common/images/location-generic-on.svg')
    }
    &.is-pending {
      background-image: none;
    }
  }
  &-useMyLocation {
    padding: 15px 10px 0 0;
  }
  &-btn {
    position: relative;
    padding: 0 0 0 20px;
    color: #333333;
    letter-spacing: -0.25px;
    text-decoration: underline;
  }
  &-btn:before {
    content: "";
    position: absolute;
    left: 0;
    top: 50%;
    height: 16px;
    width: 16px;
    transform: translateY(-50%);
    background-image: url('public/common/images/icon_geolocation.svg');
    background-repeat: no-repeat;
    background-size: cover;
    background-position: 50% 50%;
  }
}


@keyframes fadeOut {
  /* postcss-bem-linter: ignore */
  0% { 
    background-color:#e6e6e6;
  }
  /* postcss-bem-linter: ignore */
  100% {
    background-color: null;
  }
}