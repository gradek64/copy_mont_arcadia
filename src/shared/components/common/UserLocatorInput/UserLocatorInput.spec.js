import testComponentHelper from 'test/unit/helpers/test-component'
import UserLocatorInput from './UserLocatorInput'
import debounce from 'lodash.debounce'

jest.mock('lodash.debounce', () => jest.fn((f) => f))

describe('<UserLocatorInput', () => {
  const predictionArray = [
    {
      place_id: '1',
      description: 'Somewhere',
    },
    {
      place_id: '2',
      description: 'Somewhere',
    },
    {
      place_id: '3',
      description: 'Somewhere',
    },
  ]

  const props = {
    onGetCurrentPositionSuccess: jest.fn(),
    getCurrentLocationError: jest.fn(),
    setFormField: jest.fn(),
    fetchAutocomplete: jest.fn(),
    setSelectedPlace: jest.fn(),
    resetPredictions: jest.fn(),
    passPlaceDescriptionToStoreFinder: jest.fn(),
    resetSearchTerm: jest.fn(),
    resetSelectedPlace: jest.fn(),
    setUserLocatorPending: jest.fn(),
    predictions: [],
    userLocatorSearch: null,
    selectedPlaceDetails: {},
    selectedCountry: null,
    pending: false,
    setUserInputGeoLocation: jest.fn(),
    clearGeolocationError: jest.fn(),
    geoLocationError: false,
  }

  const renderComponent = testComponentHelper(UserLocatorInput.WrappedComponent)

  describe('@renders', () => {
    it('in default state', () => {
      expect(renderComponent(props).getTree()).toMatchSnapshot()
    })
    it('has predictions', () => {
      expect(
        renderComponent({
          ...props,
          predictions: predictionArray,
        }).getTree()
      ).toMatchSnapshot()
    })
    it('has geolocation errors', () => {
      expect(
        renderComponent({
          ...props,
          geoLocationError: 'Random geolocation',
        }).getTree()
      ).toMatchSnapshot()
    })
    it('has geolocation', () => {
      expect(
        renderComponent({
          ...props,
          isFeatureStoreLocatorGpsEnabled: true,
          country: 'a',
          selectedCountry: 'a',
        }).getTree()
      ).toMatchSnapshot()
    })
    it('has geolocation pending state', () => {
      expect(
        renderComponent({
          ...props,
          isFeatureStoreLocatorGpsEnabled: true,
          country: 'a',
          selectedCountry: 'a',
          pending: true,
        }).getTree()
      ).toMatchSnapshot()
    })
    it('selectedCountry=defaultCountry should show input field', () => {
      expect(
        renderComponent({
          ...props,
          selectedCountry: 'United Kingdom',
        }).getTree()
      ).toMatchSnapshot()
    })
    it('has userLocatorSearch', () => {
      expect(
        renderComponent({
          ...props,
          userLocatorSearch: 'searchy mcSearchFace',
        }).getTree()
      ).toMatchSnapshot()
    })
    it('no userLocatorSearch and pending', () => {
      expect(
        renderComponent({
          ...props,
          pending: true,
        }).getTree()
      ).toMatchSnapshot()
    })
    it('has selectedPlaceDetails', () => {
      expect(
        renderComponent({
          ...props,
          selectedPlaceDetails: {
            place_id: 122,
          },
        }).getTree()
      ).toMatchSnapshot()
    })
    it('enabled submit button if submitAlwaysEnabled', () => {
      const { wrapper, getTreeFor } = renderComponent({
        ...props,
        submitAlwaysEnabled: true,
      })
      expect(
        getTreeFor(wrapper.find('.UserLocator-goButton'))
      ).toMatchSnapshot()
    })
  })

  describe('@events', () => {
    beforeEach(() => {
      jest.clearAllMocks()
    })

    it('should resetPredictions via documentClick method', () => {
      const { instance } = renderComponent(props)
      expect(instance.props.resetPredictions).toHaveBeenCalledTimes(0)
      instance.onDocumentClick()
      expect(instance.props.resetPredictions).toHaveBeenCalledTimes(1)
    })

    describe('input is visible', () => {
      describe('input onchange', () => {
        it('should call correct props on change of input', () => {
          const { instance, wrapper } = renderComponent({
            ...props,
            selectedCountry: 'United Kingdom',
            userLocatorSearch: 'searchy mcSearchFace',
            geoLocationError: 'Random geolocation error',
          })

          expect(instance.props.resetSelectedPlace).toHaveBeenCalledTimes(0)
          expect(instance.props.setFormField).toHaveBeenCalledTimes(0)
          expect(instance.props.fetchAutocomplete).toHaveBeenCalledTimes(0)
          expect(debounce).toHaveBeenCalledTimes(1)

          wrapper
            .find('input')
            .simulate('change', { target: { value: 'blah' } })

          expect(instance.props.clearGeolocationError).toHaveBeenCalledTimes(1)
          expect(instance.props.resetSelectedPlace).toHaveBeenCalledTimes(1)
          expect(instance.props.setFormField).toHaveBeenCalledTimes(1)
          expect(instance.props.setFormField).toHaveBeenCalledWith(
            'userLocator',
            'userLocation',
            'blah'
          )
          expect(instance.props.fetchAutocomplete).toHaveBeenCalledTimes(1)
          expect(instance.props.fetchAutocomplete).toHaveBeenCalledWith('blah')
        })

        it('should clear value if current input value is Current Location', () => {
          const { instance, wrapper } = renderComponent({
            ...props,
            country: 'United Kingdom',
            isFeatureStoreLocatorGpsEnabled: true,
            selectedCountry: 'United Kingdom',
            userLocatorSearch: 'Current Location',
            userInputGeoLocation: true,
          })

          const locationInput = wrapper.find('#UserLocatorInput')

          expect(locationInput.props().value).toBe('Current Location')

          locationInput.simulate('click')

          expect(instance.props.setFormField).toHaveBeenCalledWith(
            'userLocator',
            'userLocation',
            ''
          )
        })

        it('should focus but not clear input value if not Current Location', () => {
          const { instance, wrapper } = renderComponent({
            ...props,
            country: 'United Kingdom',
            isFeatureStoreLocatorGpsEnabled: true,
            selectedCountry: 'United Kingdom',
            userLocatorSearch: 'W1T 3NL',
          })

          const locationInput = wrapper.find('#UserLocatorInput')

          expect(locationInput.props().value).toBe('W1T 3NL')

          locationInput.simulate('click')

          expect(instance.props.setFormField).not.toHaveBeenCalled()
        })
      })

      describe('clear search button', () => {
        it('clear button', () => {
          const { instance, wrapper } = renderComponent({
            ...props,
            selectedCountry: 'United Kingdom',
            country: 'United Kingdom',
            userLocatorSearch: 'searchy mcSearchFace',
          })

          expect(instance.props.setFormField).toHaveBeenCalledTimes(0)
          expect(instance.props.resetPredictions).toHaveBeenCalledTimes(0)
          expect(instance.props.resetSearchTerm).toHaveBeenCalledTimes(0)
          expect(instance.props.resetSelectedPlace).toHaveBeenCalledTimes(0)

          wrapper.find('.UserLocatorInput-clearButton').simulate('click')

          expect(instance.props.setFormField).toHaveBeenCalledTimes(1)
          expect(instance.props.setFormField).toHaveBeenCalledWith(
            'userLocator',
            'userLocation',
            ''
          )
          expect(instance.props.resetPredictions).toHaveBeenCalledTimes(1)
          expect(instance.props.resetSearchTerm).toHaveBeenCalledTimes(1)
          expect(instance.props.resetSelectedPlace).toHaveBeenCalledTimes(1)
        })
      })
    })

    describe('Get my current location button', () => {
      let preventDefault
      let getCurrentPos

      beforeEach(() => {
        preventDefault = jest.fn()
        getCurrentPos = jest.fn()

        global.window.navigator.geolocation = {
          getCurrentPosition: getCurrentPos,
        }
      })

      afterEach(() => {
        global.window.navigator.geolocation = undefined
      })

      it('should call preventDefault when clicked', () => {
        const { wrapper } = renderComponent({
          ...props,
          geoLocationEnabled: true,
          country: 'a',
          selectedCountry: 'a',
          isFeatureStoreLocatorGpsEnabled: true,
        })

        expect(preventDefault).toHaveBeenCalledTimes(0)
        wrapper
          .find('.UserLocatorInput-currentLocationButton')
          .simulate('click', { preventDefault })
        expect(preventDefault).toHaveBeenCalledTimes(1)
      })

      it('should call setUserLocatorPending', () => {
        const { instance, wrapper } = renderComponent({
          ...props,
          geoLocationEnabled: true,
          country: 'a',
          selectedCountry: 'a',
          isFeatureStoreLocatorGpsEnabled: true,
        })

        expect(instance.props.setUserLocatorPending).toHaveBeenCalledTimes(0)

        wrapper
          .find('.UserLocatorInput-currentLocationButton')
          .simulate('click', { preventDefault })

        expect(instance.props.setUserLocatorPending).toHaveBeenCalledTimes(1)
        expect(instance.props.setUserLocatorPending).toHaveBeenCalledWith(true)
      })

      it('should call getCurrentPosition with correct arguments', () => {
        const { wrapper } = renderComponent({
          ...props,
          geoLocationEnabled: true,
          country: 'a',
          selectedCountry: 'a',
          isFeatureStoreLocatorGpsEnabled: true,
        })

        expect(getCurrentPos).toHaveBeenCalledTimes(0)

        wrapper
          .find('.UserLocatorInput-currentLocationButton')
          .simulate('click', { preventDefault })

        expect(getCurrentPos).toHaveBeenCalledTimes(1)
      })
    })

    describe('prediction list', () => {
      it('should call correct methods on click', () => {
        const { instance } = renderComponent({
          ...props,
          selectedCountry: 'United Kingdom',
          userLocatorSearch: 'searchy mcSearchFace',
          predictions: predictionArray,
        })

        expect(instance.props.resetPredictions).toHaveBeenCalledTimes(0)
        expect(instance.props.setSelectedPlace).toHaveBeenCalledTimes(0)
        expect(instance.props.setFormField).toHaveBeenCalledTimes(0)
        expect(
          instance.props.passPlaceDescriptionToStoreFinder
        ).toHaveBeenCalledTimes(0)

        instance.setUserSelectedPlace(predictionArray[0])

        expect(instance.props.resetPredictions).toHaveBeenCalledTimes(1)
        expect(instance.props.setSelectedPlace).toHaveBeenCalledTimes(1)
        expect(instance.props.setFormField).toHaveBeenCalledTimes(1)
      })
      it('should call correct methods on arrow key/enter select of prediction', () => {
        const { wrapper, instance } = renderComponent({
          ...props,
          selectedCountry: 'United Kingdom',
          userLocatorSearch: 'searchy mcSearchFace',
          predictions: predictionArray,
        })

        expect(instance.props.resetPredictions).toHaveBeenCalledTimes(0)
        expect(instance.props.setSelectedPlace).toHaveBeenCalledTimes(0)
        expect(instance.props.setFormField).toHaveBeenCalledTimes(0)

        const locationInput = wrapper.find('#UserLocatorInput')
        locationInput.simulate('keydown', { keyCode: 40 })
        locationInput.simulate('keydown', { keyCode: 13 })

        expect(instance.props.resetPredictions).toHaveBeenCalledTimes(1)
        expect(instance.props.setSelectedPlace).toHaveBeenCalledWith({
          place_id: '2',
          description: 'Somewhere',
        })
        expect(instance.props.setFormField).toHaveBeenCalledTimes(1)
      })
    })
  })
})
