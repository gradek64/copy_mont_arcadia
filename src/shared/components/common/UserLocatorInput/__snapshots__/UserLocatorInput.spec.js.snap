// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserLocatorInput @renders enabled submit button if submitAlwaysEnabled 1`] = `
<Button
  className="UserLocator-goButton"
  isDisabled={true}
  title="Go"
  type="submit"
>
  Go
</Button>
`;

exports[`<UserLocatorInput @renders has geolocation 1`] = `
<div
  className="UserLocatorInput"
>
  <div
    className="UserLocatorInput-inputContainer"
  >
    <div
      className="UserLocatorInput-queryInput"
    >
      <label
        className="screen-reader-text"
        htmlFor="UserLocatorInput"
      >
        Enter town/postcode
      </label>
      <input
        autoComplete="off"
        className="
                UserLocatorInput-inputField
                is-enabled
              "
        id="UserLocatorInput"
        name="searchTerm"
        onChange={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        placeholder="Enter town/postcode"
        type="search"
        value={null}
      />
      <div
        className="
                UserLocatorInput-suffix
                
                is-enabled
              "
      >
        <button
          className="
                    UserLocatorInput-rightIcon
                    UserLocatorInput-currentLocationButton
                    
                    
                  "
          onClick={[Function]}
          title="Get my current location"
          type="button"
        >
          <span
            className="screen-reader-text"
          >
            Get my current location
          </span>
        </button>
      </div>
    </div>
    <Button
      className="UserLocator-goButton"
      isDisabled={true}
      title="Go"
      type="submit"
    >
      Go
    </Button>
  </div>
</div>
`;

exports[`<UserLocatorInput @renders has geolocation errors 1`] = `
<div
  className="UserLocatorInput"
>
  <div
    className="UserLocatorInput-inputContainer"
  >
    <div
      className="UserLocatorInput-queryInput"
    >
      <label
        className="screen-reader-text"
        htmlFor="UserLocatorInput"
      >
        Enter town/postcode
      </label>
      <input
        autoComplete="off"
        className="
                UserLocatorInput-inputField
                
              "
        id="UserLocatorInput"
        name="searchTerm"
        onChange={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        placeholder="Enter town/postcode"
        type="search"
        value={null}
      />
      <div
        className="
                UserLocatorInput-suffix
                
                
              "
      />
    </div>
    <Button
      className="UserLocator-goButton"
      isDisabled={true}
      title="Go"
      type="submit"
    >
      Go
    </Button>
  </div>
  <div
    className="UserLocator-error"
  >
    <p>
      Random geolocation
    </p>
    <span
      className="screen-reader-text"
    >
      Random geolocation
    </span>
  </div>
</div>
`;

exports[`<UserLocatorInput @renders has geolocation pending state 1`] = `
<div
  className="UserLocatorInput"
>
  <div
    className="UserLocatorInput-inputContainer"
  >
    <div
      className="UserLocatorInput-queryInput"
    >
      <label
        className="screen-reader-text"
        htmlFor="UserLocatorInput"
      >
        Enter town/postcode
      </label>
      <input
        autoComplete="off"
        className="
                UserLocatorInput-inputField
                is-enabled
              "
        id="UserLocatorInput"
        name="searchTerm"
        onChange={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        placeholder="Enter town/postcode"
        type="search"
        value={null}
      />
      <div
        className="
                UserLocatorInput-suffix
                
                is-enabled
              "
      >
        <button
          className="
                    UserLocatorInput-rightIcon
                    UserLocatorInput-currentLocationButton
                    is-pending
                    
                  "
          onClick={[Function]}
          title="Get my current location"
          type="button"
        >
          <Loader
            className="UserLocatorInput-loader"
            fillColor="#000"
            isButton={false}
            isInInput={true}
          />
          <span
            className="screen-reader-text"
          >
            Get my current location
          </span>
        </button>
      </div>
    </div>
    <Button
      className="UserLocator-goButton"
      isDisabled={true}
      title="Go"
      type="submit"
    >
      Go
    </Button>
  </div>
</div>
`;

exports[`<UserLocatorInput @renders has predictions 1`] = `
<div
  className="UserLocatorInput"
>
  <div
    className="UserLocatorInput-inputContainer is-predictionsListOpen"
  >
    <div
      className="UserLocatorInput-queryInput"
    >
      <label
        className="screen-reader-text"
        htmlFor="UserLocatorInput"
      >
        Enter town/postcode
      </label>
      <input
        autoComplete="off"
        className="
                UserLocatorInput-inputField
                
              "
        id="UserLocatorInput"
        name="searchTerm"
        onChange={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        placeholder="Enter town/postcode"
        type="search"
        value={null}
      />
      <div
        className="
                UserLocatorInput-suffix
                
                
              "
      />
    </div>
    <Button
      className="UserLocator-goButton"
      isDisabled={true}
      title="Go"
      type="submit"
    >
      Go
    </Button>
    <ul
      className="UserLocatorInput-predictionsList"
    >
      <li
        className="UserLocatorInput-predictionsListItem UserLocatorInput-active"
        onClick={[Function]}
        role="presentation"
      >
        Somewhere
      </li>
      <li
        className="UserLocatorInput-predictionsListItem UserLocatorInput-inActive"
        onClick={[Function]}
        role="presentation"
      >
        Somewhere
      </li>
      <li
        className="UserLocatorInput-predictionsListItem UserLocatorInput-inActive"
        onClick={[Function]}
        role="presentation"
      >
        Somewhere
      </li>
    </ul>
  </div>
</div>
`;

exports[`<UserLocatorInput @renders has selectedPlaceDetails 1`] = `
<div
  className="UserLocatorInput"
>
  <div
    className="UserLocatorInput-inputContainer"
  >
    <div
      className="UserLocatorInput-queryInput"
    >
      <label
        className="screen-reader-text"
        htmlFor="UserLocatorInput"
      >
        Enter town/postcode
      </label>
      <input
        autoComplete="off"
        className="
                UserLocatorInput-inputField
                
              "
        id="UserLocatorInput"
        name="searchTerm"
        onChange={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        placeholder="Enter town/postcode"
        type="search"
        value={null}
      />
      <div
        className="
                UserLocatorInput-suffix
                
                
              "
      />
    </div>
    <Button
      className="UserLocator-goButton"
      isDisabled={true}
      title="Go"
      type="submit"
    >
      Go
    </Button>
  </div>
</div>
`;

exports[`<UserLocatorInput @renders has userLocatorSearch 1`] = `
<div
  className="UserLocatorInput"
>
  <div
    className="UserLocatorInput-inputContainer"
  >
    <div
      className="UserLocatorInput-queryInput"
    >
      <label
        className="screen-reader-text"
        htmlFor="UserLocatorInput"
      >
        Enter town/postcode
      </label>
      <input
        autoComplete="off"
        className="
                UserLocatorInput-inputField
                
              "
        id="UserLocatorInput"
        name="searchTerm"
        onChange={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        placeholder="Enter town/postcode"
        type="search"
        value="searchy mcSearchFace"
      />
      <div
        className="
                UserLocatorInput-suffix
                has-text
                
              "
      >
        <button
          className="
                    UserLocatorInput-rightIcon
                    UserLocatorInput-clearButton
                    
                  "
          onClick={[Function]}
          title="Clear"
          type="button"
        >
          <span
            className="screen-reader-text"
          >
            Clear
          </span>
        </button>
      </div>
    </div>
    <Button
      className="UserLocator-goButton"
      isDisabled={true}
      title="Go"
      type="submit"
    >
      Go
    </Button>
  </div>
</div>
`;

exports[`<UserLocatorInput @renders in default state 1`] = `
<div
  className="UserLocatorInput"
>
  <div
    className="UserLocatorInput-inputContainer"
  >
    <div
      className="UserLocatorInput-queryInput"
    >
      <label
        className="screen-reader-text"
        htmlFor="UserLocatorInput"
      >
        Enter town/postcode
      </label>
      <input
        autoComplete="off"
        className="
                UserLocatorInput-inputField
                
              "
        id="UserLocatorInput"
        name="searchTerm"
        onChange={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        placeholder="Enter town/postcode"
        type="search"
        value={null}
      />
      <div
        className="
                UserLocatorInput-suffix
                
                
              "
      />
    </div>
    <Button
      className="UserLocator-goButton"
      isDisabled={true}
      title="Go"
      type="submit"
    >
      Go
    </Button>
  </div>
</div>
`;

exports[`<UserLocatorInput @renders no userLocatorSearch and pending 1`] = `
<div
  className="UserLocatorInput"
>
  <div
    className="UserLocatorInput-inputContainer"
  >
    <div
      className="UserLocatorInput-queryInput"
    >
      <label
        className="screen-reader-text"
        htmlFor="UserLocatorInput"
      >
        Enter town/postcode
      </label>
      <input
        autoComplete="off"
        className="
                UserLocatorInput-inputField
                
              "
        id="UserLocatorInput"
        name="searchTerm"
        onChange={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        placeholder="Enter town/postcode"
        type="search"
        value={null}
      />
      <div
        className="
                UserLocatorInput-suffix
                
                
              "
      />
    </div>
    <Button
      className="UserLocator-goButton"
      isDisabled={true}
      title="Go"
      type="submit"
    >
      Go
    </Button>
  </div>
</div>
`;

exports[`<UserLocatorInput @renders selectedCountry=defaultCountry should show input field 1`] = `
<div
  className="UserLocatorInput"
>
  <div
    className="UserLocatorInput-inputContainer is-visible"
  >
    <div
      className="UserLocatorInput-queryInput"
    >
      <label
        className="screen-reader-text"
        htmlFor="UserLocatorInput"
      >
        Enter town/postcode
      </label>
      <input
        autoComplete="off"
        className="
                UserLocatorInput-inputField
                
              "
        id="UserLocatorInput"
        name="searchTerm"
        onChange={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        placeholder="Enter town/postcode"
        type="search"
        value={null}
      />
      <div
        className="
                UserLocatorInput-suffix
                
                
              "
      />
    </div>
    <Button
      className="UserLocator-goButton"
      isDisabled={true}
      title="Go"
      type="submit"
    >
      Go
    </Button>
  </div>
</div>
`;
